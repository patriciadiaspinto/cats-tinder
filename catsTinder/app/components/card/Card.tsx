import React, {useEffect} from 'react';
import {
  View,
  StyleSheet,
  Platform,
  Dimensions,
  Image,
  Text,
} from 'react-native';

const {width, height} = Dimensions.get('window');

const Card = ({data}: any) => {
  return (
    data && (
      <View style={[styles.photoWrapper, styles.shadow]} testID={'container'}>
        <Image source={{uri: data.url}} style={[styles.photo]} />
        {data.breeds?.length > 0 ? (
          <View style={styles.dataWrapper}>
            <View style={styles.data}>
              <Text style={styles.dataFont} testID={'name'}>
                {data.breeds[0].name}
              </Text>
              <Text style={styles.dataFont} testID={'life_span'}>
                {data.breeds[0].life_span}
              </Text>
            </View>
            <Text style={styles.description} testID={'origin'}>
              {data.breeds[0].origin}
            </Text>
          </View>
        ) : null}
      </View>
    )
  );
};

const styles = StyleSheet.create({
  shadow: {
    shadowColor: '#52006A',
    ...Platform.select({
      ios: {
        shadowOffset: {width: -2, height: 4},
        shadowOpacity: 0.2,
        shadowRadius: 3,
      },
      android: {
        elevation: 20,
      },
    }),
  },
  photo: {
    width: width * 0.9,
    height: height * 0.55,
    borderRadius: 16,
    position: 'relative',
  },
  photoWrapper: {
    alignItems: 'center',
    borderRadius: 16,
    marginTop: 32,
  },
  dataWrapper: {
    display: 'flex',
    flexDirection: 'column',
    borderTopRightRadius: 16,
    borderTopLeftRadius: 16,
    position: 'absolute',
    backgroundColor: 'white',
    bottom: 0,
    paddingHorizontal: 15,
    paddingVertical: 10,
    width: '90%',
  },
  data: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 5,
  },
  dataFont: {
    fontSize: 16,
    fontWeight: '700',
    color: '#434141',
  },
  description: {
    fontSize: 8,
    fontWeight: '700',
    color: '#BFBFC0',
    lineHeight: 10,
  },
});

export default Card;
