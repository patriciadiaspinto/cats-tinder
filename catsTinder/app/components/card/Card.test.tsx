import {render} from '@testing-library/react-native';
import Card from './Card';

const data = {
  id: 'test 1',
  url: '',
  width: 300,
  height: 600,
  breeds: [{name: 'test', life_span: '15-16', origin: 'Portugal'}],
  favorite: {},
};

describe('Card component', () => {
  it('should render Card name, life span and origin', async () => {
    const {getByTestId} = render(<Card data={data} />);

    expect(getByTestId('container')).toBeDefined;
    expect(getByTestId('name').props.children).toBe(data.breeds[0].name);
    expect(getByTestId('life_span').props.children).toBe(
      data.breeds[0].life_span,
    );
    expect(getByTestId('origin').props.children).toBe(data.breeds[0].origin);
  });

  it('should render Card', async () => {
    const tree = render(<Card data={data} />);

    expect(tree).toMatchSnapshot();
  });
});
