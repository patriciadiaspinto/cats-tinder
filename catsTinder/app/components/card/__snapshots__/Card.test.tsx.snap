// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render Card 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 16,
        "marginTop": 32,
      },
      {
        "shadowColor": "#52006A",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
      },
    ]
  }
  testID="container"
>
  <Image
    source={
      {
        "uri": "",
      }
    }
    style={
      [
        {
          "borderRadius": 16,
          "height": 733.7,
          "position": "relative",
          "width": 675,
        },
      ]
    }
  />
  <View
    style={
      {
        "backgroundColor": "white",
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "bottom": 0,
        "display": "flex",
        "flexDirection": "column",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "position": "absolute",
        "width": "90%",
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#434141",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
        testID="name"
      >
        test
      </Text>
      <Text
        style={
          {
            "color": "#434141",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
        testID="life_span"
      >
        15-16
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#BFBFC0",
          "fontSize": 8,
          "fontWeight": "700",
          "lineHeight": 10,
        }
      }
      testID="origin"
    >
      Portugal
    </Text>
  </View>
</View>
`;
